$NetBSD: patch-otherlibs_threads_Makefile,v 1.10 2017/09/10 08:56:47 wiz Exp $

Honor LDFLAGS.
Use BSD_INSTALL_* instead of mkdir and cp

--- otherlibs/threads/Makefile.orig	2017-07-13 08:56:44.000000000 +0000
+++ otherlibs/threads/Makefile
@@ -33,6 +33,8 @@ CAML_OBJS=thread.cmo mutex.cmo condition
 
 LIB=../../stdlib
 
+LDOPTS=-ldopt "$(LDFLAGS)"
+
 LIB_OBJS=$(LIB)/camlinternalFormatBasics.cmo pervasives.cmo		\
   $(LIB)/array.cmo $(LIB)/list.cmo $(LIB)/char.cmo $(LIB)/bytes.cmo	\
   $(LIB)/string.cmo $(LIB)/sys.cmo $(LIB)/sort.cmo marshal.cmo		\
@@ -59,16 +61,16 @@ all: libvmthreads.a threads.cma stdlib.c
 allopt:
 
 libvmthreads.a: $(C_OBJS)
-	$(MKLIB) -o threads -oc vmthreads $(C_OBJS)
+	$(MKLIB) -o threads -oc vmthreads $(C_OBJS) $(LDOPTS)
 
 threads.cma: $(CAML_OBJS)
-	$(MKLIB) -ocamlc '$(CAMLC)' -o threads -oc vmthreads $(CAML_OBJS)
+	$(MKLIB) -ocamlc '$(CAMLC)' -o threads -oc vmthreads $(CAML_OBJS) $(LDOPTS)
 
 stdlib.cma: $(LIB_OBJS)
 	$(CAMLC) -a -o stdlib.cma $(LIB_OBJS)
 
 unix.cma: $(UNIXLIB_OBJS)
-	$(MKLIB) -ocamlc '$(CAMLC)' -o unix -linkall $(UNIXLIB_OBJS)
+	$(MKLIB) -ocamlc '$(CAMLC)' -o unix -linkall $(UNIXLIB_OBJS) $(LDOPTS)
 
 pervasives.cmo: pervasives.mli pervasives.cmi pervasives.ml
 	$(CAMLC) ${COMPFLAGS} -nopervasives -c pervasives.ml
@@ -111,12 +113,12 @@ CMIFILES=thread.cmi mutex.cmi condition.
 
 install:
 	if test -f dllvmthreads.so; then \
-	  cp dllvmthreads.so $(INSTALL_STUBLIBDIR)/.; \
+	  $(BSD_INSTALL_LIB) dllvmthreads.so $(INSTALL_STUBLIBDIR)/.; \
 	fi
-	mkdir -p $(INSTALL_LIBDIR)/vmthreads
-	cp libvmthreads.a $(INSTALL_LIBDIR)/vmthreads/libvmthreads.a
+	$(BSD_INSTALL_LIB_DIR) $(INSTALL_LIBDIR)/vmthreads
+	$(BSD_INSTALL_DATA) libvmthreads.a $(INSTALL_LIBDIR)/vmthreads/libvmthreads.a
 	cd $(INSTALL_LIBDIR)/vmthreads; $(RANLIB) libvmthreads.a
-	cp $(CMIFILES) $(CMIFILES:.cmi=.mli) $(CMIFILES:.cmi=.cmti) \
+	$(BSD_INSTALL_DATA) $(CMIFILES) $(CMIFILES:.cmi=.mli) $(CMIFILES:.cmi=.cmti) \
 	   threads.cma stdlib.cma unix.cma $(INSTALL_LIBDIR)/vmthreads
 
 installopt:
